### PR TITLE
Issue/#83 release annotations

### PR DESCRIPTION
**Pull request by korzol**
Hello @ceckoslab 
Here is latest changes required to achieve dynamic release annotations

Kindly review and let me know

P.S. Also need to update test to reflect latest changes

### CODE REVIEW

**Comment by korzol** [comment]
Hello @ceckoslab 
I fixed currently existing tests and prepared to implement dynamic releases tests. I guess we need some tests for that.
Also I think we need to implement releases for other types of diagram, am I right? 
So I hope tomorrow we will have tests and releases implemented for all kind of diagrams

**Comment by ceckoslab** [comment]
Hello @korzol 

I think that it looks fine for now. Please do not implement more things for other diagram types because I am planning to do some refactoring around the view logic.

Could you go in to details about what and where you plan to implement more tests?

**Comment by korzol** [comment]
Hello @ceckoslab 
I thought about adding another file into /tests/BasicRum/DiagramBuilder/ directory. Lets say DevicePerformaceTest.php or DevicePerformaceDesktopTest.php which will test diagram building for device performance (desktop) which is time_series and where dynamic releases is applicable

**Comment by ceckoslab** [comment]
Hello @korzol 

Could you write one more when the Data Layer returns empty result set? I remember that we were getting a notice for undefined variable in this case and our tests were failing.

**Comment by ceckoslab** [comment]
Hello @korzol 

Is this PR still a WIP? If not, could you remove the WIP part of the title?

File: tests/BasicRum/DiagramBuilder/FirstByteAndBounceRateTest.php [hunk 1]
```diff
@@ -2,61 +2,69 @@
...
 
-use PHPUnit\Framework\TestCase;
-
 use App\BasicRum\DiagramBuilder;
 use App\BasicRum\DiagramOrchestrator;
+use App\BasicRum\Release;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
-class FirstByteAndBounceRateTest extends TestCase
+class FirstByteAndBounceRateTest extends KernelTestCase
```
[the remark this answers is not in the document]
Comment (korzol) [reply tests/BasicRum/DiagramBuilder/FirstByteAndBounceRateTest.php]: Fixed KernelTestCase to FixtureTestCase. 
As of phpfixer, yes obviously pre-commit hook is calling the phpfixer for all files that are about to be committed

File: tests/BasicRum/DiagramBuilder/SegmentsDeviceDistributionTest.php [hunk 2]
```diff
@@ -2,13 +2,20 @@
...
 
-use PHPUnit\Framework\TestCase;
-
 use App\BasicRum\DiagramBuilder;
 use App\BasicRum\DiagramOrchestrator;
+use App\BasicRum\Release;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
-class SegmentsDeviceDistributionTest extends TestCase
+class SegmentsDeviceDistributionTest extends KernelTestCase
```
[the remark this answers is not in the document]
Comment (korzol) [reply tests/BasicRum/DiagramBuilder/SegmentsDeviceDistributionTest.php]: Fixed as well

File: src/BasicRum/Release.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * @param string $date1 Start date
+     * @param string $date2 End date
+     *
+     * @return array [description]
+     */
+    public function getAllReleasesBetweenDates(string $date1, string $date2): array
+    {
+        $dateArray['startDate'] = new  DateTime($date1);
```
Comment (ceckoslab) [inline src/BasicRum/Release.php:26]: It doesn't make sense to use $dateArray here. You can just pass 2 arguments to findAllBetweenDates(new  DateTime($date1),  DateTime($date2)). Creating an array doesn't bring much value but increases the complexity a bit.

File: src/BasicRum/Release.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        $this->releases = $releases;
+    }
+
+    /**
+     * @param string $date1 Start date
+     * @param string $date2 End date
+     *
+     * @return array [description]
+     */
+    public function getAllReleasesBetweenDates(string $date1, string $date2): array
```
Comment (ceckoslab) [inline src/BasicRum/Release.php:24]: Please give more descriptive variables names $date1 and $date2 is not bad but what stops us to have $startData and $endDate ?

**Pull request merged**